src/lib/price.ts

```diff
@@ -57,6 +58,44 @@ export async function getTokenPrices(tokens: Token[]): Promise<PricesResponse> {
   return fetchTokenPrices(Array.from(keys))
 }
 
+export async function tokensBalancesWithPrices(tokens: Token[]) {
```

Comment:
why not use `getPricedBalances` ?

Reply:
Originally I tried using it but then switched to this. Since `getPricedBalances` does checks for rewards and underlyings which (I believe?) are not needed for tokens balances handler, it added more time to compute. So I made this fn specifically for the tokens balances handler

Reply:
it only adds a few `if` statements